### DISCUSSION TITLE
Parcel v2 generates several JS files which cannot be ran by node

### DISCUSSION THREAD

**Discussion by declanelcocks** [parcel-bundler/parcel#9255] [Q&A]
In the past I was using Parcel v1 along with `parcel-plugin-run-server` to build a `dist` output of a TypeScript Node.js backend and run it. The options were quite simple:

```
const options = {
  outDir: './dist',
  outFile: 'main.js',
  publicUrl: './',
  watch: true,
  cache: true,
  cacheDir: '.cache',
  contentHash: false,
  minify: false,
  scopeHoist: false,
  target: 'node',
  bundleNodeModules: false,
  logLevel: 3,
  sourceMaps: true,
  detailedReport: false,
  autoInstall: true,
};
```

This would generate some static files and one, big `dist/main.js` file that the plugin could run using `node dist/main.js`. I've tried to copy these options across to Parcel v2 with:

```
{
    entries: entryFiles,
    shouldDisableCache: false,
    shouldContentHash: false,
    shouldAutoInstall: true,
    mode: 'development',
    logLevel: 'error',
    targets: {
      default: {
        context: 'node',
        publicUrl: './',
        distDir: './dist',
        distEntry: 'main.js',
        optimize: false,
        scopeHoist: false,
        includeNodeModules: false,
        sourceMap: true,
      },
    },
  }
```

This will build `dist/main.js` but it will also build several other files like `main.5c9552c8.js`. This itself is fine, I guess it's code splitting the bundles as the docs explain, but when running `node dist/main.js` it now results in this error:

```
node:internal/modules/cjs/loader:1026
  const err = new Error(message);
              ^

Error: Cannot find module '9vinu'
Require stack:
- /Users/<REPO_PATH>/dist/main.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1026:15)
    at Function.Module._load (node:internal/modules/cjs/loader:871:27)
    at Module.require (node:internal/modules/cjs/loader:1098:19)
    at newRequire (/Users/<REPO_PATH>/dist/main.js:58:18)
    at newRequire (/Users/<REPO_PATH>/dist/main.b8135af0.js:53:18)
    at newRequire (/Users/<REPO_PATH>/dist/main.b8135af0.js:45:18)
    at localRequire (/Users/<REPO_PATH>/dist/main.b8135af0.js:84:35)
    at Object.aE43U.a48566ca5e2ae079 (/Users/<REPO_PATH>/dist/main.b8135af0.js:151:20)
    at newRequire (/Users/<REPO_PATH>/dist/main.b8135af0.js:71:24)
    at /Users/<REPO_PATH>/dist/main.b8135af0.js:122:5 {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/Users/<REPO_PATH>/dist/main.js' ]
}
```

Can I configure Parcel to output a single JS file? Or is there something else I'm missing?

Thanks 🙏 

(Seems similar to https://github.com/parcel-bundler/parcel/issues/5683)

**Comment by mischnic** [comment]
Try adding `outputFormat: "commonjs"` to the `default: {..}` object, not sure if that is already what's being used though.

**Reply by declanelcocks** [reply]
@mischnic Yeah same error unfortunately. I had the same thought myself initially so tried out the various output formats. I believe `commonjs` is the default option here if you don't define it.
